presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/UnwrapCastInComparison.java

```diff
@@ -309,7 +332,7 @@ private Expression unwrapCast(ComparisonExpression expression)
                         return TRUE_LITERAL;
                     case LESS_THAN:
                     case LESS_THAN_OR_EQUAL:
-                        if (sourceRange.isPresent() && compare(sourceType, sourceRange.get().getMin(), literalInSourceType) == 0) {
+                        if (sourceRange.isPresent() && compare(sourceType, sourceRange.get().getMin(), literalInSourceType) == 0 && implicitInjectiveCoercionAbility == FULL) {
```

Comment:
The additional condition is redundant. For PARTIAL, the allowed range for literal here is well within source type range. Instead, please add a comment.